design/0001-frr.md

```diff
@@ -0,0 +1,340 @@
+│               │  │                    │               │  └─┤  <<package>>  │
+│     nodes     │  │                    │   Protocol    │    │               │
+│    <<Node>>   ├──┘                    │ <<interface>> │    └───────────────┘
+│               │                       │               │
+└───────────────┘                       └───────▲───────┘
+                                                │<<implements>>
+                                     ┌──────────┴─────────┐
+                                     │                    │
+                              ┌──────┴────────┐  ┌────────┴──────┐
+                              │     main::    │  │    main::     │
+                              │ L2_controller │  │ bgp_controller│
+                              │   <<class>>   │  │   <<class>>   │
+                              │               │  │               │
+                              └───────────────┘  └───────┬───────┘
+                                                         │
+                                                         │  <<use>>
+                                                 ┌───────▼───────┐
+                                                 │               │
+                                                 │    session    │
+                                                 │ <<interface>> │
+                                                 │               │
+                                                 └───────▲───────┘
+                                                         │ <<implement>>
+                                             ┌───────────┴─────────────┐
+                                             │                         │
+                                     ┌───────┴───────┐         ┌───────┴───────┐
+                                     │               │         │               │
+                                     │  bgp_metallb  │         │    bgp_frr    │
+                                     │  <<package>>  │         │  <<package>>  │
+                                     │               │         │               │
+                                     └───────────────┘         └───────┬───────┘
+                                                                       │
+                                                                       │
+                                                               ┌───────┴───────┐
+                                                               │               │
+                                                               │      frr      │
+                                                               │ <<container>> │
+                                                               │               │
+                                                               └───────────────┘
+
+```
+
+#### FRR deployment
+
+FRR will be deployed in a container as part of the speaker Pod. This will
```

Comment:
I am not sure what the resource utilization is but I can't expect it would be to high. Therefore I would prefer on all setups.

When idle, you can see the utilization here which is very low:
```
USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
frr      2186725  0.0  0.0 329756  5728 ?        S<sl 06:36   0:00 /usr/lib/frr/zebra -d -A 127.0.0.1
frr      2186735  0.0  0.0 259736  7532 ?        S<sl 06:36   0:00 /usr/lib/frr/bgpd -d -A 127.0.0.1
root     2186749  0.0  0.0  80824  3448 ?        S<s  06:36   0:00 /usr/lib/frr/watchfrr -d -r /usr/lib/frr/frr restart %s -s /usr/lib/frr/frr start %s -k /usr/lib/frr/frr stop %s zebra bgpd
```

Reply:
Ok, let's go with that and we can adjust if needed. Let's hope the requests for cpu/mem for that container don't need to be too big (not for the idle case you show, for the case where it is indeed used).